packages/interchain-token-service-api/src/primitives.rs

```diff
@@ -0,0 +1,90 @@
+use crate::error::Error;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::Report;
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct TokenId {
+    #[serde(with = "axelar_wasm_std::hex")]
+    #[schemars(with = "String")]
+    pub id: [u8; 32],
+}
```

Comment:
consider just `pub type TokenId = [u8;32];`

Reply:
since TokenId will be a public facing struct, we'd like to serialize it as hex. That's not doable if it's a type alias, right?